pkg/pool-stable-phantom/test/StablePhantomPool.test.ts

```diff
@@ -759,6 +759,75 @@ describe('StablePhantomPool', () => {
         await expect(pool.exit({ data: wrongUserData })).to.be.revertedWith('Transaction reverted without a reason');
       });
 
+      describe('exit BPT in for one token out', () => {
+        let tokenIndex: number;
+        let token: Token;
+
+        sharedBeforeEach('get token to exit with', async () => {
+          tokenIndex = pool.bptIndex == 0 ? 1 : 0;
+          token = tokens.get(tokenIndex);
+        });
+
+        context('not in recovery mode', () => {
+          itExitsExactBptInForOneTokenOutProperly();
+        });
+
+        context('in recovery mode', () => {
+          sharedBeforeEach('enable recovery mode', async () => {
+            await pool.enableRecoveryMode(admin);
+          });
+
+          itExitsExactBptInForOneTokenOutProperly();
+        });
+
+        function itExitsExactBptInForOneTokenOutProperly() {
+          it('grants one token for exact bpt', async () => {
+            // 20% of previous balance
+            const previousBptBalance = await pool.balanceOf(lp);
+            const bptIn = pct(previousBptBalance, 0.2);
+            const expectedTokenOut = await pool.estimateTokenOutGivenBptIn(token, bptIn);
+
+            const result = await pool.singleExitGivenIn({ from: lp, bptIn, token });
+
+            // Only token out should be the one transferred
+            expect(result.amountsOut[tokenIndex]).to.be.equalWithError(expectedTokenOut, 0.0001);
+            expect(result.amountsOut.filter((_, i) => i != tokenIndex)).to.be.zeros;
+
+            const bptAfter = await pool.balanceOf(lp);
+
+            // Current BPT balance should decrease
+            expect(previousBptBalance.sub(bptIn)).to.equal(bptAfter);
+          });
+
+          it('can tell token amount returned (for either exit or exitSwap)', async () => {
+            const bptIn = pct(await pool.balanceOf(lp), 0.2);
+            const expectedTokenOut = await pool.estimateTokenOutGivenBptIn(token, bptIn);
+
+            const result = await pool.querySingleExitGivenIn({ bptIn, token });
+
+            expect(result.bptIn).to.equal(bptIn);
+            expect(result.amountsOut.filter((_, i) => i != tokenIndex)).to.be.zeros;
+            expect(result.amountsOut[tokenIndex]).to.be.equalWithError(expectedTokenOut, 0.0001);
```

Comment:
This test doesn't really do much, in that it is the same test as before (comparing the solidity implementation with the TS one), except using queries instead of actual joins. For completeness we could test that the query returns the same result as the actual operation, but that'd be a different test (i.e. do the query, then do the thing, and check for exact equality).

Reply:
How could it not, though? By definition they run the same code. And if A=B and A=C, then B=C.